File: packages/subject-segmentation/ios/Plugin/SubjectSegmentationPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(SubjectSegmentationPlugin, "SubjectSegmentation",
           CAP_PLUGIN_METHOD(processImage, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(isGoogleSubjectSegmentationModuleAvailable, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(installGoogleSubjectSegmentationModule, CAPPluginReturnPromise);
)

File: packages/subject-segmentation/ios/Plugin/SubjectSegmentationPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SubjectSegmentationPlugin)
public class SubjectSegmentationPlugin: CAPPlugin {
    private let implementation = SubjectSegmentation()

    @objc func processImage(_ call: CAPPluginCall) {
        rejectCallAsUnimplemented(call)
    }

    @objc func isGoogleSubjectSegmentationModuleAvailable(_ call: CAPPluginCall) {
        rejectCallAsUnimplemented(call)
    }

    @objc func installGoogleSubjectSegmentationModule(_ call: CAPPluginCall) {
        rejectCallAsUnimplemented(call)
    }

    private func rejectCallAsUnavailable(_ call: CAPPluginCall) {
        call.unavailable("This method is not available on this platform.")
    }

    private func rejectCallAsUnimplemented(_ call: CAPPluginCall) {
        call.unimplemented("This method is not available on this platform.")
    }
}
